pinot-controller/src/main/java/org/apache/pinot/controller/recommender/io/InputManager.java

```diff
@@ -0,0 +1,519 @@
+import org.apache.pinot.controller.recommender.rules.RulesToExecute;
+import org.apache.pinot.controller.recommender.rules.io.params.BloomFilterRuleParams;
+import org.apache.pinot.controller.recommender.rules.io.params.FlagQueryRuleParams;
+import org.apache.pinot.controller.recommender.rules.io.params.InvertedSortedIndexJointRuleParams;
+import org.apache.pinot.controller.recommender.rules.io.params.NoDictionaryOnHeapDictionaryJointRuleParams;
+import org.apache.pinot.controller.recommender.rules.io.params.PartitionRuleParams;
+import org.apache.pinot.controller.recommender.rules.utils.FixedLenBitset;
+import org.apache.pinot.spi.data.DimensionFieldSpec;
+import org.apache.pinot.spi.data.FieldSpec;
+import org.apache.pinot.spi.data.MetricFieldSpec;
+import org.apache.pinot.spi.data.Schema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static java.lang.Math.max;
+import static org.apache.pinot.controller.recommender.rules.io.params.RecommenderConstants.*;
+
+
+@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.NONE)
+public class InputManager {
+  private final Logger LOGGER = LoggerFactory.getLogger(InputManager.class);
+
+  /******************************Deserialized from input json*********************************/
+  // Basic input fields
+  public RulesToExecute _rulesToExecute = new RulesToExecute(); // dictates which rules to execute
+  public Schema _schema = new Schema();
+  public SchemaWithMetaData _schemaWithMetaData = new SchemaWithMetaData();
+
+  public String _queryType = SQL; // SQL or PQL
+  public long _qps = DEFAULT_QPS;
+  public Map<String, Double> _queryWeightMap = new HashMap<>(); // {"queryString":"queryWeight"}
+  public String _tableType = OFFLINE;
+  public long _numMessagesPerSec = DEFAULT_NUM_MSG_PER_SEC; // messages per sec for kafka to consume
+  public long _numRecordsPerPush = DEFAULT_NUM_RECORDS_PER_PUSH; // records per push for offline part of a table
+  public long _latencySLA = DEFAULT_LATENCY_SLA; // latency sla in ms
+  public int _numKafkaPartitions = DEFAULT_NUM_KAFKA_PARTITIONS;
+
+  // The parameters of rules
+  public PartitionRuleParams _partitionRuleParams = new PartitionRuleParams();
+  public InvertedSortedIndexJointRuleParams _invertedSortedIndexJointRuleParams =
+      new InvertedSortedIndexJointRuleParams();
+  public BloomFilterRuleParams _bloomFilterRuleParams = new BloomFilterRuleParams();
+  public NoDictionaryOnHeapDictionaryJointRuleParams _noDictionaryOnHeapDictionaryJointRuleParams =
+      new NoDictionaryOnHeapDictionaryJointRuleParams();
+  public FlagQueryRuleParams _flagQueryRuleParams = new FlagQueryRuleParams();
+
+  // For forward compatibility: 1. dev/sre to overwrite field(s) 2. incremental recommendation on existing/staging tables
+  public ConfigManager _overWrittenConfigs = new ConfigManager();
+
+  /******************************Ignored by deserializer****************************************/
```

Comment:
I don't think I fully understand why this should be ignored by the deserializer

Reply:
The fields following this line are pre-processed from the input and used as algorithm input only, so no need to serialize/deserialize them in any place

Reply:
Got it. Thanks.